src/bin/tikv-server.rs

```diff
@@ -158,14 +158,13 @@ fn run_raft_server(pd_client: RpcClient, cfg: &TiKvConfig, security_mgr: Arc<Sec
             let pd_sender = pd_sender.clone();
             move || storage::ReadPoolContext::new(pd_sender.clone())
         });
-    let mut storage = create_raft_storage(raft_router.clone(), &cfg.storage, storage_read_pool)
-        .unwrap_or_else(|e| fatal!("failed to create raft stroage: {:?}", e));
-    storage
-        .mut_gc_worker()
-        .set_local_storage(Arc::clone(&kv_engine));
-    storage
-        .mut_gc_worker()
-        .set_raft_store_router(raft_router.clone());
+    let storage = create_raft_storage(
+        raft_router.clone(),
+        &cfg.storage,
+        storage_read_pool,
+        Some(Arc::clone(&kv_engine)),
+        Some(raft_router.clone()),
+    ).unwrap_or_else(|e| fatal!("failed to create raft stroage: {:?}", e));
```

Comment:
Hmmm..  I think it is not *very* necessary because this signature is rarely used. Mostly the builder style is used and the argument list is very clean.